### ISSUE TITLE
dfu mode not recognised

### ISSUE THREAD

**Issue by phongshader** [makerbase-mks/MKS-RUMBA32#12]
mac:
FS mode
<img width="244" alt="Screen Shot 2020-06-08 at 7 46 12 PM" src="https://user-images.githubusercontent.com/38082624/84100599-17b8f100-a9c1-11ea-968b-f14a82874c4c.png">
dfu mode
<img width="240" alt="Screen Shot 2020-06-08 at 7 45 25 PM" src="https://user-images.githubusercontent.com/38082624/84100585-0d96f280-a9c1-11ea-987d-597fb31b94fb.png">

Win 10
VCP_V1.5.0_Setup_W8_x64_64bits.exe installed, it recognized the board worked once to get firmware onto board but I can no longer update the firmware.
FS mode:
![FS](https://user-images.githubusercontent.com/38082624/84101038-fb698400-a9c1-11ea-91f6-6a66a354aef6.PNG)
dfu mode
![dfu](https://user-images.githubusercontent.com/38082624/84101068-0c19fa00-a9c2-11ea-95cb-f21a4a9a9ff9.png)
no STM32 bootloader.
Is there a way to load firmware without dfu?

platformio:
`Uploading .pio/build/rumba32_mks/firmware.bin
dfu-util 0.9

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2020 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Match vendor ID from file: 0483
Match product ID from file: df11
dfu-util: No DFU capable USB device available
*** [upload] Error 74`

**Comment by makerbase-mks** [comment]
Hi! Have you viewed MKS RUMBA32 wiki? The wiki Has Tutorial about how to use MKS RUMBA32 and so on

**Comment by phongshader** [comment]
I pulled out an ooold win7 netbook and it recognized the board in dfu mode. I was able to load firmware...it took several hours to compile. After that I was able to connect using a newer laptop. I don't know what happened but it doesn't seem to be a board issue

**Comment by Ratisse** [comment]
Had same DFU problem, spent 2 days to find a solution:

Disconnect the board from USB
Press and hold BOOT button (don't touch the reset button)
Connect USB cable
Release Boot button
Normally, in device manager, you will see STM32 BOOTLOADER and will be able to upload your firmware. This worked for me (MKS Rumba32). Don't know why the original procedure disconnect completly the card

Hope this help

**Comment by phongshader** [comment]
After working for 5 months I'm back to the same problem. I've tried several different laptops, Mac and Windows both unable to connect in DFU mode. I tried the above procedure with no success. I think the boot button is no longer functional or at least it seems to have no affect on the board when it is pressed.

**Comment by mks-viva** [comment]
Abou DFU mode : https://github.com/makerbase-mks/MKS-RUMBA32/wiki/How-to-upload-marlin-firmware

**Comment by Ratisse** [comment]
Power on the board and check the MCU temperature with your finger, if it is very hot, probably it has a problem. Mine had this issue (I could sometimes enter DFU mode but not always and had other issues after loading the firmware). I had to buy another board to compare, the new board MCU is cold and it works properly after loading the firmware (entering DFU mode works everytime).

**Comment by phongshader** [comment]
I have tried to connect in DFU mode when the board is cold...let it sit overnight, so overheating is not an issue. I've gone through the link posted above and the boot button does nothing when pressed. When pressed in combination with the reset button the board just disconnects and never shows up in the device manager, the boot loader driver does not fail, it never tries to load because there is no device present.

**Comment by mks-viva** [comment]
@Ratisse 
So, it looks more like a CPU failure?

**Comment by mks-viva** [comment]
@phongshader 
Before this, your motherboard was working properly?

**Comment by phongshader** [comment]
The board will still run the printer fine, I just can't update the firmware anymore. What happened was I made some changes to the firmware loaded it in DFU mode, ran some test prints, the changes I made did not fix the issue I was having, made some changes to the firmware, recompiled, and when I tried to load the new firmware I got the error "dfu-util: No DFU capable USB device available". I tried a dozen different times on different usb ports on a macbook pro, and 2 different win 10 laptops. The board shows up fine in win10 and macOS but as soon as I try to go into DFU mode it disappears. Yes it was working properly.

**Comment by mks-viva** [comment]
@phongshader 
That is, the motherboard is still normal, but the motherboard cannot enter DFU mode? Can i understand this way.

**Comment by phongshader** [comment]
Yes the motherboard still functions but is not recognized or doesn't enter DFU mode when trying to upload firmware.

**Comment by mks-viva** [comment]
So, MKS RUMBA32 wiki about 'how to uopload firmware' can it be helpful for you?

**Comment by phongshader** [comment]
the problem is that the device is not seen by either macOS or windows so "We need to show STM32 BOOTLOADER installation failed" will never happen because the laptop sees no STM32 BOOTLOADER. Zadig does not see it either.

**Comment by DrumClock** [comment]
Hi @mks-viva , @makerbase-mks 
first I loaded FW Marlin using STM32 BOOTLOADER and DFU mode without any problems. 
Now I wanted to upload a new FW but nothing is happening.
WIN10 Device Manager does not see the STM32 BOOTLOADER.
If I connect the board it is not a problem to communicate and send commands via USB and PRONTERFACE.

How do I load FW on a board when I don't see STM32 BOOTLOADER in the manager?

**Comment by DrumClock** [comment]
Hi,
I tried another computer and it's the same.
Everything indicates that after holding the BOOT button and plugging in the USB cable the board does not invoke the DFU mod.

If I reset the board with the RESET button, it connects to the computer as COM5 
and the pronterface communicates with it without any problems.

This situation when it is not possible to run the DFU mod occurred after about 2 to 3 flash of FW Marlin.

**Comment by DrumClock** [comment]
Heureka!
**the inability to start the DFU mode is caused by the connected EXP2, EXP1 connector with TFT35 E3 V1.0 display.
If theEXP1 EXP2 diisconnects the DFU mod is activated.**

**Comment by PADYT0007** [comment]
HELLO I AM HAVING A PROBLEM WITH MY RUMBA 32 ... I RECEIVED AN ERROR .. ALL THE INSTRUCTIONS ARE FOLLOWED
I USE WINDOWS 11
![image](https://user-images.githubusercontent.com/116293119/196993474-895f0855-b140-4405-a6b2-5d98e6b7597c.png)
![image](https://user-images.githubusercontent.com/116293119/196994931-a4a5cc37-40bd-4f8b-a711-efffba7ce36c.png)
![image](https://user-images.githubusercontent.com/116293119/196995069-bd911b60-473e-43ad-b6d4-6b381cdb9f2e.png)
PLS HELP!!!!!!!!!!!

**Comment by DrumClock** [comment]
Hi, @PADYT0007  I don't use Merlin anymore, I switched to the better FW Klipper.
So I can't for you  do nothing.

**Comment by PADYT0007** [comment]
> Hi, @PADYT0007 I don't use Merlin anymore, I switched to the better FW Klipper.
> So I can't for you do nothing.
Can I use this FW Klipper for the rumba 32 v 1.0

**Comment by DrumClock** [comment]
> > Hi, @PADYT0007 I don't use Merlin anymore, I switched to the better FW Klipper.
> > So I can't for you do nothing.
> > Can I use this FW Klipper for the rumba 32 v 1.0

Klipper is installed on Raspberry Pi, the rumba board basically serves only as a terminal for controlling the printer. read the documentation at https://www.klipper3d.org
